zeus/device/gpu/amd.py

```diff
@@ -241,29 +251,49 @@ def getAverageMemoryPowerUsage(self) -> int:
         )
 
     @_handle_amdsmi_errors
-    def supportsGetTotalEnergyConsumption(self) -> bool:
-        """Check if the GPU supports retrieving total energy consumption."""
-        if self._supportsGetTotalEnergyConsumption is None:
-            try:
-                _ = amdsmi.amdsmi_get_energy_count(self.handle)
-                self._supportsGetTotalEnergyConsumption = True
-            except amdsmi.AmdSmiLibraryException as e:
-                if (
-                    e.get_error_code() == 2
-                ):  # amdsmi.amdsmi_wrapper.AMDSMI_STATUS_NOT_SUPPORTED
-                    self._supportsGetTotalEnergyConsumption = False
-                else:
-                    raise e
-
+    def supportsGetTotalEnergyConsumption(
+        self,
+    ) -> bool:
+        """Check if the GPU supports retrieving total energy consumption. Returns a future object of the result."""
+        wait_time = 0.5  # seconds
+        threshold = 0.8  # 80% threshold
```

Comment:
Do we want to first check whether `self._supportsGetTotalEnergyConsumption is not None` and return the cached value for future invocations of this method?

Reply:
with the current structure it's never called more than once. But that would be better, I'll add it

Reply:
This method is part of the public API of `GPU` and `GPUs`, meaning people can always call it in their code. Thanks!